src/Plugin/StrawberryRunnersPostProcessor/OcrPostProcessor.php

```diff
@@ -278,51 +295,103 @@ public function run(\stdClass $io, $context = StrawberryRunnersPostProcessorPlug
-      // assume its not there/won't work.
-      $proc_output_check_searchable = 0;
-
-      if ($execstring_check_searchable) {
-        $backup_locale = setlocale(LC_CTYPE, '0');
-        setlocale(LC_CTYPE, $backup_locale);
-        // Support UTF-8 commands.
-        // @see http://www.php.net/manual/en/function.shell-exec.php#85095
-        shell_exec("LANG=en_US.utf-8");
-        $proc_output_check_searchable = $this->proc_execute($execstring_check_searchable, $timeout);
-      }
-
+      //Also with ALTO: PrintSpace element doesn't have children (ComposedBlock AND/OR TextBlock) if blank or not searchable
+      //pdfalto -noLineNumbers -noImage -noImageInline -readingOrder -f 1 -l 1 page1.pdf - |grep Block |wc -l (return 0 OR 1)
+      //
+      //To test switch from hOCR to ALTO
+      $formatocr = 'ALTO';
 
-      if ($proc_output_check_searchable == 1) {
+      if ($formatocr == 'HOCR') {
+        setlocale(LC_CTYPE, 'en_US.UTF-8');
+        $execstring_check_searchable = $this->buildExecutableCommand_checkSearchable($io);
+        // assume its not there/won't work.
+        $proc_output_check_searchable = 0;
 
-        //if searchable run djvu2hocr
-        //
+        if ($execstring_check_searchable) {
+          $backup_locale = setlocale(LC_CTYPE, '0');
+          setlocale(LC_CTYPE, $backup_locale);
+          // Support UTF-8 commands.
+          // @see http://www.php.net/manual/en/function.shell-exec.php#85095
+          shell_exec("LANG=en_US.utf-8");
+          $proc_output_check_searchable = $this->proc_execute($execstring_check_searchable, $timeout);
+        }
+      }
+      else {
         setlocale(LC_CTYPE, 'en_US.UTF-8');
-        $execstring_djvu2hocr = $this->buildExecutableCommand_djvu2hocr($io);
-        if ($execstring_djvu2hocr) {
+        $execstring_check_searchable = $this->buildExecutableCommand_pdfalto($io) . "|grep Block |wc -l";
+        // assume its not there/won't work.
+        $proc_output_check_searchable = 0;
+
+        if ($execstring_check_searchable) {
           $backup_locale = setlocale(LC_CTYPE, '0');
           setlocale(LC_CTYPE, $backup_locale);
           // Support UTF-8 commands.
           // @see http://www.php.net/manual/en/function.shell-exec.php#85095
           shell_exec("LANG=en_US.utf-8");
-          $proc_output = $this->proc_execute($execstring_djvu2hocr, $timeout);
-          if (is_null($proc_output)) {
-            $this->logger->warning("DJVU processing via {$execstring_djvu2hocr} timed out");
-            throw new \Exception("Could not execute {$execstring_djvu2hocr} or timed out");
-          }
+          $proc_output_check_searchable = $this->proc_execute($execstring_check_searchable, $timeout);
+        }
+      }
+
+      if ($proc_output_check_searchable > 0) {
 
-          //djvu2hocr output uses ocrx_line while tesseract uses ocr_line
+        //To test switch from hOCR to ALTO
+        $formatocr = 'ALTO';
```

Comment:
@giancarlobi hi! Is this for testing? Or are you forcing ALTO as default?

Reply:
@DiegoPino it's a my trivial way to force to use ALTO preserving originally hocr code to compare them developing. Also at line 302 above. This can be used if we want make user able to choose between HOCR or ALTO, not sure if this makes sense becuase with ALTO working I don't see any reasons to choose hocr.